cpp/src/arrow/flight/types.h

```diff
@@ -834,5 +837,147 @@ class ARROW_FLIGHT_EXPORT SimpleResultStream : public ResultStream {
   size_t position_;
 };
 
+/// \defgroup flight-error Error Handling
+/// Types for handling errors from RPCs.  Flight uses a set of status
+/// codes standardized across Flight implementations, so these types
+/// let applications work directly with those codes instead of having
+/// to translate to and from Arrow Status.
+/// @{
+
+/// \brief Abstract status code for an RPC as per the Flight
+///   specification.
+enum class TransportStatusCode {
```

Comment:
Wild idea: instead of reinventing another set of status codes, how about using HTTP status codes? They are well-known and fine-grained enough for our purposes. They will also map naturally to many backends.

Reply:
Hmm, we'd have to juggle how they map with gRPC, but that may be reasonable

Reply:
Note that this is *not* a new set of codes, these are just refactored out of the existing implementation.

Reply:
And I'd prefer to explore that separately from this, if possible

Reply:
No pb from me :-)

Reply:
There's already a mapping: https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md

It loses a lot of information, though (it's also meant for HTTP -> gRPC, not the other way around)

Supposedly Protobuf (or some Google API spec) might define something more comprehensive: https://github.com/cloudendpoints/esp/blob/1b02c011f071a8d400dbbbc08b512700b69ae519/src/api_manager/utils/status.cc#L287-L327

I filed https://github.com/apache/arrow/issues/36722